apps/web/src/app/(internal)/%5F%5Fhtmltopdf/certificate/page.tsx

```diff
@@ -93,40 +90,30 @@ export default async function SigningCertificate({ searchParams }: SigningCertif
       return 'Unknown';
     }
 
-    const documentAuthOptions = ZDocumentAuthOptionsSchema.parse(document.authOptions);
-    const recipientAuthOptions = ZRecipientAuthOptionsSchema.parse(recipient.authOptions);
+    const extractedAuthMethods = extractDocumentAuthMethods({
+      documentAuth: document.authOptions,
+      recipientAuth: recipient.authOptions,
+    });
 
     let authLevel = 'Email';
 
-    if (
-      documentAuthOptions.globalAccessAuth === 'ACCOUNT' ||
-      recipientAuthOptions.accessAuth === 'ACCOUNT'
-    ) {
+    if (extractedAuthMethods.derivedRecipientAccessAuth === 'ACCOUNT') {
       authLevel = 'Account Authentication';
     }
 
-    if (
-      documentAuthOptions.globalActionAuth === 'ACCOUNT' ||
-      recipientAuthOptions.actionAuth === 'ACCOUNT'
-    ) {
+    if (extractedAuthMethods.derivedRecipientActionAuth === 'ACCOUNT') {
       authLevel = 'Account Re-Authentication';
     }
 
-    if (
-      documentAuthOptions.globalActionAuth === 'TWO_FACTOR_AUTH' ||
-      recipientAuthOptions.actionAuth === 'TWO_FACTOR_AUTH'
-    ) {
-      authLevel = 'Two Factor Re-Authentication';
+    if (extractedAuthMethods.derivedRecipientActionAuth === 'TWO_FACTOR_AUTH') {
+      authLevel = 'Two-Factor Re-Authentication';
     }
 
-    if (
-      documentAuthOptions.globalActionAuth === 'PASSKEY' ||
-      recipientAuthOptions.actionAuth === 'PASSKEY'
-    ) {
+    if (extractedAuthMethods.derivedRecipientActionAuth === 'PASSKEY') {
       authLevel = 'Passkey Re-Authentication';
     }
 
-    if (recipientAuthOptions.actionAuth === 'EXPLICIT_NONE') {
+    if (extractedAuthMethods.derivedRecipientActionAuth === 'EXPLICIT_NONE') {
       authLevel = 'Email';
     }
```

Comment:
Is it possible to throw a tsmatch in or something so when we add new action auth methods we can cause build errors?

Reply:
Yeah that should work, for some reason I thought it wouldn't but it should do

Reply:
Although it wouldn't be exhaustive since we consider access and action auth 🤔

Reply:
You could create a function which has tspattern inside that takes in the actionAuth and returns a string or null right

If it's null you use the default authLevel email/account in this outer function

Reply:
Aight